src/main/java/org/sagebionetworks/bridge/services/PermissionService.java

```diff
@@ -0,0 +1,169 @@
+        this.assessmentService = assessmentService;
+    }
+    
+    // For mock testing
+    protected String generateGuid() {
+        return BridgeUtils.generateGuid();
+    }
+    
+    public PermissionDetail createPermission(String appId, Permission permission) {
+        checkArgument(isNotBlank(appId));
+        checkNotNull(permission);
+        
+        permission.setGuid(generateGuid());
+        
+        Validate.entityThrowingException(INSTANCE, permission);
+        
+        Permission createdPermission = permissionDao.createPermission(appId, permission);
+        
+        return getPermissionDetail(appId, createdPermission);
+    }
+    
+    public PermissionDetail updatePermission(String appId, Permission permission) {
+        checkArgument(isNotBlank(appId));
+        checkNotNull(permission);
+    
+        Validate.entityThrowingException(INSTANCE, permission);
+        
+        Permission existingPermission = permissionDao.getPermission(appId, permission.getGuid())
+                .orElseThrow(() -> new EntityNotFoundException(Permission.class));
+        
+        // Can only update accessLevel
+        permission.setAppId(existingPermission.getAppId());
+        permission.setUserId(existingPermission.getUserId());
+        permission.setEntityType(existingPermission.getEntityType());
+        permission.setEntityId(existingPermission.getEntityId());
+        
+        Permission updatedPermission = permissionDao.updatePermission(appId, permission);
+    
+        return getPermissionDetail(appId, updatedPermission);
+    }
+    
+    public List<PermissionDetail> getPermissionsForUser(String appId, String userId) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(userId));
+    
+        List<Permission> permissions = permissionDao.getPermissionsForUser(appId, userId);
+    
+        List<PermissionDetail> permissionDetails = new ArrayList<>();
+        for (Permission permission : permissions) {
+            permissionDetails.add(getPermissionDetail(appId, permission));
+        }
+    
+        return permissionDetails;
+    }
+    
+    public List<PermissionDetail> getPermissionsForEntity(String appId, String entityType, String entityId) {
+        checkArgument(isNotBlank(appId));
+        checkNotNull(entityType);
+        checkArgument(isNotBlank(entityId));
+        
+        List<Permission> permissions = permissionDao.getPermissionsForEntity(appId, EntityType.valueOf(entityType), entityId);
+        
+        List<PermissionDetail> permissionDetails = new ArrayList<>();
+        for (Permission permission : permissions) {
+            permissionDetails.add(getPermissionDetail(appId, permission));
+        }
+        
+        return permissionDetails;
+    }
+    
+    public void deletePermission(String appId, String guid) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(guid));
+        
+        permissionDao.deletePermission(appId, guid);
```

Comment:
What happens if the GUID doesn't exist? (Should throw ENFE)

Reply:
Added a get call to check if the permission exists first.